### ISSUE TITLE
Tabs tray: Use icon from Session and observe changes

### ISSUE THREAD

**Issue by pocmo** [mozilla-mobile/fenix#3846]
Follow-up from the "missing icons" investigation here: https://github.com/mozilla-mobile/android-components/issues/3294

One reason (there are others to fix in AC) for icons not showing up in the tabs tray is that Fenix is not using the icon from `Session` if it is available. Also since they are loaded asynchronously Fenix should observe Session icon changes to display the icon in cases where the load completes while the tabs tray is showing.

Basically the algorithm for displaying an icon in the tabs tray should be:

* (1) If `Session.icon` is not `null` then display this icon.
* (2) Else: Use `BrowserIcons` to load an icon for the current URL
* (3) On `Session.Observer.onIconChanged()` getting invoked use the updated icon passed to the observer or just start at (1).

**Comment by ekager** [comment]
Closing because I think we are now following that checklist for updating the tabs and we will QA in https://github.com/mozilla-mobile/fenix/issues/5351
